src/init/features/functions/typescript.js

```diff
@@ -64,3 +65,11 @@ module.exports = function (setup, config) {
       return npmDependencies.askInstallDependencies(setup.functions, config);
     });
 };
+
+function getPackage(useLint) {
+  const nodeEngineVersion = utils.getNodeVersionString();
```

Comment:
When do we ever check that the user's node version is one supported by Google Cloud Functions?

Reply:
@taeold I was thinking of using [`isDeprecatedRuntime()`](https://github.com/firebase/firebase-tools/blob/f413eb9eb6940ee20dea748b18bb25ce185e7d7f/src/deploy/functions/runtimes/index.ts#L22) to check for deprecated node version and [`utils.logWarning()`](https://github.com/firebase/firebase-tools/blob/f413eb9eb6940ee20dea748b18bb25ce185e7d7f/src/utils.ts#L174) to display the warning message. Something like this:

```js
if (isDeprecatedRuntime(`nodejs${nodeEngineVersion}`)) {
    utils.logWarning("Your Node version is not supported in Google Cloud Functions.");
}
```

If it's good enough, I'll go ahead and update the code.

Reply:
👍🏼  Tiny bug I'd watch out for is that something like `nodejs13`, which is not supported by Google Cloud Functions, may pass the `isDeprecatedRuntime()` check. You might to throw in `isValidRuntime` check somewhere in the code.